pkg/linter/lintRuleNonRelevantCIDRCommon.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package linter
+
+import (
+	"fmt"
+
+	"github.com/np-guard/models/pkg/ipblock"
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/vpcmodel"
+)
+
+// a rule in which the src/dst (depending on if ingress or egress) referring to the vpc is disjoint to it or
+// has a part to within the vpc and is not all range
+type ruleNonRelevantCIDR struct {
+	rule        vpcmodel.RuleOfFilter
+	vpcResource vpcmodel.VPC
+	disjoint    bool // is the relevant src/dst block disjoint to the VPC address range;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////
+// functionality used by both SG and NACL lints
+////////////////////////////////////////////////////////////////////////////////////////////
+
+func findRuleNonRelevantCIDR(configs map[string]*vpcmodel.VPCConfig, filterLayerName string) (res []ruleNonRelevantCIDR, err error) {
+	for _, config := range configs {
+		if config.IsMultipleVPCsConfig {
+			continue // no use in executing lint on dummy vpcs
+		}
+		vpcAddressRange := config.VPC.AddressRange()
+		filterLayer := config.GetFilterTrafficResourceOfKind(filterLayerName)
+		rules, err := filterLayer.GetRules()
+		if err != nil {
+			return nil, err
+		}
+		// for ingress dst addresses ips within the VPC, for egress src
+		for _, rule := range rules {
```

Comment:
```suggestion
		for i := range rules {
			rule := &rules[i]
```
Avoid copying structs

Reply:
right. I'll do it in the remaining place as well

Reply:
done in all relevant places